### PR TITLE
Bdshot for AP_Peiph and Matek L431

### PR DESCRIPTION
**Pull request by andyp1per**
This implements bdshot for the Matek L431 AP_Periph device. It means that ESC RPM information can be retrieved from the attached ESCs without requiring a telemetry wire. It also means faster update rates which can be controlled via ```ESC_TELEM_RATE```.

This PR also allows ```OUT_DSHOT_RATE``` to be used to increase the servo update rate and this achieve faster backend telemetry rates.

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32L431xx.py [hunk 1]
```diff
@@ -31,7 +31,8 @@
     # flags of 2 means faster memory for CPU intensive work
     'RAM_MAP' : [
         # use 2nd mapping of SRAM2 to allow a single memory space
-        (0x20000000, 64, 1), # SRAM1/SRAM2
+        (0x20000000, 48, 0), # SRAM1
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32L431xx.py:34]: please check if it works without this change

Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32L431xx.py:34]: may need ALT_RAM_MAP

**Review by tridge — CHANGES_REQUESTED** [review]
need to delay to after 4.3 is branched

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32L431xx.py (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32L431xx.py]: Removed

**Comment by tridge** [comment]
testing on F4, F7 would be great, thanks @andyp1per !

**Comment by andyp1per** [comment]
Flew on a bunch of bdshot platforms - all good

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**